crates/common/src/opa.rs

```diff
@@ -280,6 +292,104 @@ impl PolicyLoader for CliPolicyLoader {
     }
 }
 
+#[derive(Clone, Default)]
+pub struct UrlPolicyLoader {
+    policy_id: String,
+    address: String,
+    policy: Vec<u8>,
+    entrypoint: String,
+}
+
+impl UrlPolicyLoader {
+    pub fn new(url: &str, policy_id: &str, entrypoint: &str) -> Self {
+        Self {
+            address: url.into(),
+            policy_id: policy_id.to_owned(),
+            entrypoint: entrypoint.to_owned(),
+            ..Default::default()
+        }
+    }
+
+    async fn fetch_bundle_from_url(&mut self) -> Result<Vec<u8>, PolicyLoaderError> {
+        let response = reqwest::get(&self.address).await?;
+        if response.status().is_success() {
+            let bytes = response.bytes().await?;
+            Ok(bytes.to_vec())
+        } else {
+            Err(PolicyLoaderError::HttpError(response.status().to_string()))
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl PolicyLoader for UrlPolicyLoader {
+    fn set_address(&mut self, address: &str) {
+        self.address = address.to_owned();
+    }
+
+    fn set_rule_name(&mut self, name: &str) {
+        self.policy_id = name.to_owned();
+    }
+
+    fn set_entrypoint(&mut self, entrypoint: &str) {
+        self.entrypoint = entrypoint.to_owned();
+    }
+
+    fn get_address(&self) -> &str {
+        &self.address
+    }
+
+    fn get_rule_name(&self) -> &str {
+        &self.policy_id
+    }
+
+    fn get_entrypoint(&self) -> &str {
+        &self.entrypoint
+    }
+
+    fn get_policy(&self) -> &[u8] {
+        &self.policy
+    }
+
+    fn load_policy_from_bytes(&mut self, policy: &[u8]) {
+        self.policy = policy.to_vec();
+    }
+
+    async fn load_policy(&mut self) -> Result<(), PolicyLoaderError> {
+        let url = &self.address;
+
+        let bundle = if url.starts_with("http://")
+            || url.starts_with("https://")
+            || url.starts_with("file://")
+        {
+            self.fetch_bundle_from_url().await?
```

Comment:
Probably good to test that `reqwest` even works for file URLs.

Reply:
But how to mock a file URL was what I couldn't figure out. Is there a suitable target we can test with?

Reply:
`test_load_policy_from_bare_path` does a fine job of creating a temporary file, I'd think it should work to take a similar approach but prefix its absolute path with `file://` to construct a good test target.

Reply:
Ah, I'll give it a go. Confused by thinking a file URL refers to a file that lives online somewhere.